File: app/src/main/java/com/example/aichat/AIChatApp.java
package com.example.aichat;

import android.app.Application;

import androidx.appcompat.app.AppCompatDelegate;
import androidx.lifecycle.DefaultLifecycleObserver;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.ProcessLifecycleOwner;

import com.mikepenz.iconics.Iconics;

public class AIChatApp extends Application implements DefaultLifecycleObserver {
    @Override
    public void onCreate() {
        super.onCreate();
        Iconics.init(this); // typefaces auto-register via ContentProvider
        applyTheme();
        RoomMigrationHelper.migrateIfNeeded(this);
        new ProactiveMessageNotifier(this).ensureChannel();
        ProcessLifecycleOwner.get().getLifecycle().addObserver(this);
        ProactiveMessageWorkScheduler.scheduleNext(this);
    }

    private void applyTheme() {
        String theme = new ConfigManager(this).getTheme();
        int mode = "dark".equals(theme) ? AppCompatDelegate.MODE_NIGHT_YES
                : "light".equals(theme) ? AppCompatDelegate.MODE_NIGHT_NO
                : AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM;
        AppCompatDelegate.setDefaultNightMode(mode);
    }

    @Override
    public void onStart(LifecycleOwner owner) {
        ProactiveMessageWorkScheduler.cancel(this);
    }

    @Override
    public void onStop(LifecycleOwner owner) {
        ProactiveMessageWorkScheduler.scheduleNext(this);
    }
}

File: app/src/main/java/com/example/aichat/AIChatApp.kt
package com.example.aichat

import android.app.Application
import androidx.appcompat.app.AppCompatDelegate
import com.example.aichat.story.InkosRulesMigration
import com.example.aichat.sync.RemoteSyncConfigStore
import com.example.aichat.sync.SyncScheduler
import com.example.aichat.sync.WsClient
import com.mikepenz.iconics.Iconics
import com.tom_roush.pdfbox.android.PDFBoxResourceLoader

class AIChatApp : Application() {
    override fun onCreate() {
        super.onCreate()
        Iconics.init(this) // typefaces auto-register via ContentProvider
        PDFBoxResourceLoader.init(this)
        applyTheme()
        VolcEngineTTSManager.init(this)
        VolcEngineHttpTTS.init(cacheDir)
        RoomMigrationHelper.migrateIfNeeded(this)
        // v16 后置迁移: 把 Room 暂存的旧 inkosBookRulesYaml 写成 rules outline item.
        InkosRulesMigration.runIfPending(this)
        ProactiveMessageNotifier(this).ensureChannel()
        if (RemoteSyncConfigStore(this).isEnabled()) {
            SyncScheduler.start(this)
            WsClient.start(this)
        }
    }

    private fun applyTheme() {
        val theme = ConfigManager(this).getTheme()
        val mode = when (theme) {
            "dark" -> AppCompatDelegate.MODE_NIGHT_YES
            "light" -> AppCompatDelegate.MODE_NIGHT_NO
            else -> AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM
        }
        AppCompatDelegate.setDefaultNightMode(mode)
    }
}
